dist/dialog.linker.build.js

```diff
@@ -198,14 +216,30 @@ window.jQuery = window.jQuery || window.shoestring;
 			}
 		});
 
+	// if the hash matches an ajaxlink's url, open it by triggering a click on the ajaxlink
+	$( w ).bind( "hashchange load", function(){
+		var hash = w.location.hash.split( "#" ).pop();
+		var id = hash.replace( /-dialog$/, "" );
+		var $ajaxLink = $( 'a[href="' + id +'"][data-dialog-link]' );
+		// if the link specified nohistory, don't click it
+		var nohistory = $ajaxLink.is( "[data-dialog-nohistory]" );
+		var $dialogInPage = $( '.dialog[id="' + id + '"]' );
+		if( $ajaxLink.length && !nohistory && !$dialogInPage.length ){
+			$ajaxLink.eq( 0 ).trigger( "click" );
+		}
```

Comment:
Yah I guess I was thinking that the dialog linker should extend the `open` method to create the dialog before opening it so that we can just call `open` and it will work in both cases.